### ISSUE TITLE
Received msg_key doesn't match with expected one

### ISSUE THREAD

**Issue by wind-hx** [gram-js/gramjs#35]
`
client.invoke(new functions.messages.GetDialogsRequest({
            limit: limit,
            offsetPeer: new types.InputPeerEmpty(),
        }));
`

error1: Received msg_key doesn't match with expected one
error2: Server replied with an invalid auth key
error3: Handshake inactivity timeout

How to solve these errors？

**Comment by miehilmie** [comment]
Yes, I can confirm this problem. As I dig into the root cause, it seems that if my account has only few chats, the request seems fine.

But if my account has a lot of chats, it seems that this error will occurs. Here how I send the request:

```
const response = await client.invoke(new GetDialogsRequest({
    offsetDate: last_date,
    offsetId: 0,
    offsetPeer: new InputPeerEmpty(),
    limit: chunk_size,
    hash: 0,
}))
```

The error seems to be causing by `!msgKey.equals(ourKey.slice(8, 24))` in `gramjs/network/MTProtoState.js:149` where the decrytion is failed to verify the signature is correct causing it throws `SecurityError: Received msg_key doesn't match with expected one`

This is as far as I could digest the code hope it helps identifying the actual problem. 

Thank you for this awesome project!

**Comment by painor** [comment]
this will be fixed soon. already have the code just waiting for it to be pushed

**Comment by fadakar** [comment]
please share a merge request link for this item, thanks
